core/runtime/v2/shim/util_unix.go

```diff
@@ -76,6 +83,47 @@ func SocketAddress(ctx context.Context, socketPath, id string) (string, error) {
 
 // AnonDialer returns a dialer for a socket
 func AnonDialer(address string, timeout time.Duration) (net.Conn, error) {
+	if strings.HasPrefix(address, Vsock) {
+		address = strings.TrimPrefix(address, Vsock+"://")
+		cxport := strings.Split(address, ":")
+		if len(cxport) != 2 {
+			return nil, fmt.Errorf("invalid vsock address %s", address)
+		}
+		contextID, err := strconv.ParseUint(cxport[0], 10, 0)
+		if err != nil {
+			return nil, err
+		}
+		if contextID > math.MaxUint32 {
+			return nil, fmt.Errorf("vsock context id %d is invalid", contextID)
+		}
+		port, err := strconv.ParseUint(cxport[1], 10, 0)
+		if err != nil {
+			return nil, err
+		}
+		if port > math.MaxUint32 {
+			return nil, fmt.Errorf("vsock port %d is invalid", port)
+		}
+		return vsock.Dial(uint32(contextID), uint32(port), &vsock.Config{})
+	}
+
+	// For Cloud-hypervisor, The connection is a "hybrid vsock",
```

Comment:
firecracker also supports this iirc. Might be best to just say "Some VMMs support a hybrid vsock model, ......"

Reply:
done